test/basics.js

```diff
@@ -338,4 +337,66 @@ describe('Basics', function() {
     done();
   });
 
+  it('should do requestone-get-reply', function(done) {
+    var nc = NATS.connect(PORT);
+    var initMsg = 'Hello World';
+    var replyMsg = 'Hello Back!';
+
+    nc.subscribe('foo', function(msg, reply) {
+      should.exist(msg);
+      msg.should.equal(initMsg);
+      should.exist(reply);
+      reply.should.match(/_INBOX\.*/);
+      nc.publish(reply, replyMsg);
+    });
+
+    var gotOne = false;
+    nc.requestOne('foo', initMsg, null, 1000, function(reply) {
+      should.exist(reply);
+      reply.should.equal(replyMsg);
+      if(! gotOne) {
+        gotOne = true;
+        nc.close();
+        done();
+      }
+    });
+  });
+
+  it('should do requestone-will-unsubscribe', function(done) {
+    var rsub = "x.y.z";
+    var nc = NATS.connect(PORT);
+    nc.subscribe(rsub, function(msg, reply) {
+      reply.should.match(/_INBOX\.*/);
+      nc.publish(reply, "y");
+      nc.flush();
+      setTimeout(function() {
```

Comment:
Technically, requestOne should return a single message and auto-unsubscribe automatically after first response, so you should be able to publish `y` and `z` at the same time and requestor should get only `y`.